cmd/npmExecuteLint.go

```diff
@@ -0,0 +1,139 @@
+		if ok {
+			_, ok := scripts["ci-lint"].(string)
+			if ok {
+				packagesWithCilint = append(packagesWithCilint, file)
+				log.Entry().Info("Discovered ci-lint script in " + file)
+			}
+		}
+	}
+	return packagesWithCilint
+}
+
+func runCiLint(packagesWithCilint []string, utils npm.Utils) error {
+	execRunner := utils.GetExecRunner()
+	oldWorkingDirectory, err := utils.Getwd()
+	if err != nil {
+		return fmt.Errorf("failed to get current working directory before executing ci-lint scripts: %w", err)
+	}
+
+	for _, packageJSON := range packagesWithCilint {
+		dir := path.Dir(packageJSON)
+		err := utils.Chdir(dir)
+		if err != nil {
+			return fmt.Errorf("failed to change into directory for executing ci-lint script: %w", err)
+		}
+
+		// Ignore possible errors when invoking ci-lint script to not fail the pipeline based on linting results
+		err = execRunner.RunExecutable("npm", "run", "--silent", "ci-lint")
+
+		err = utils.Chdir(oldWorkingDirectory)
+		if err != nil {
+			return fmt.Errorf("failed to change back into original directory: %w", err)
+		}
+	}
+	return nil
+}
+
+func runDefaultLint(utils npm.Utils) {
+	execRunner := utils.GetExecRunner()
+	eslintConfigs := findEslintConfigs(utils)
+
+	// If the user has ESLint configs in the project we use them to lint existing JS files. In this case we do not lint other types of files,
+	// i.e., .jsx, .ts, .tsx, since we can not be sure that the provided config enables parsing of these file types.
+	if len(eslintConfigs) > 0 {
+		for i, config := range eslintConfigs {
+			dir := path.Dir(config)
+			if dir == "." {
+				// Ignore possible errors when invoking ci-lint script to not fail the pipeline based on linting results
+				_ = execRunner.RunExecutable("npx", "eslint", ".", "-f", "checkstyle", "-o", "./"+strconv.Itoa(i)+"_defaultlint.xml", "--ignore-pattern", "node_modules/", "--ignore-pattern", ".eslintrc.js")
+			} else {
+				lintPattern := dir + "/**/*.js"
+				// Ignore possible errors when invoking ci-lint script to not fail the pipeline based on linting results
+				_ = execRunner.RunExecutable("npx", "eslint", lintPattern, "-f", "checkstyle", "-o", "./"+strconv.Itoa(i)+"_defaultlint.xml", "--ignore-pattern", "node_modules/", "--ignore-pattern", ".eslintrc.js")
+			}
+		}
+	} else {
+		// Install dependencies manually, since npx cannot resolve the dependencies required for general purpose
+		// ESLint config, e.g., TypeScript ESLint plugin
+		log.Entry().Info("Run ESLint with general purpose config")
+		// Ignore possible errors when invoking ci-lint script to not fail the pipeline based on linting results
+		_ = execRunner.RunExecutable("npm", "install", "eslint@^7.0.0", "typescript@^3.7.4", "@typescript-eslint/parser@^3.0.0", "@typescript-eslint/eslint-plugin@^3.0.0")
+		_ = execRunner.RunExecutable("npx", "--no-install", "eslint", ".", "--ext", ".js,.jsx,.ts,.tsx", "-c", ".pipeline/.eslintrc.json", "-f", "checkstyle", "-o", "./defaultlint.xml", "--ignore-pattern", ".eslintrc.js")
```

Comment:
By not using the npm package it is slightly different to the groovy implementation where executeNpm was used: 
https://github.com/SAP/cloud-s4-sdk-pipeline-lib/blob/master/vars/checkDefaultLint.groovy#L25

Thus, you are e.g. not using the download cache or custom registries the you might have defined.